packages/core/src/App/Components/Elements/Errors/error-component.jsx

```diff
@@ -1,10 +1,28 @@
 import PropTypes from 'prop-types';
 import React from 'react';
+import { useHistory } from 'react-router-dom';
 import { PageError } from '@deriv/components';
 import { routes } from '@deriv/shared';
 import { Localize } from '@deriv/translations';
 
-const ErrorComponent = ({ header, message, redirect_label, redirectOnClick, should_show_refresh = true }) => {
+const ErrorComponent = ({
+    header,
+    message,
+    redirect_label,
+    redirectOnClick,
+    with_history,
+    setError,
+    where_to = routes.trade,
+    should_show_refresh = true,
+}) => {
+    let buttonOnClick;
+    if (with_history) {
+        const history = useHistory();
```

Comment:
It's not recommended to use hooks conditionally: https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level - so you can move the useHistory one step up